privileged_residues/util.py

```diff
@@ -18,9 +18,9 @@ def apply_transform(self, xform):
     index = 0
     transformed_coords = np.dot(xform, np.stack(coords).T)
 
-    for res in self.residues:
-        for atom in res.atoms():
-            atom.xyz(V3(*transformed_coords[:3, index]))
+    for i in range(1, len(self) + 1):
+        for j in range(1, self.residue(i).natoms() + 1):
+            self.residue(i).atom(j).xyz(V3(*transformed_coords[:3, index]))
```

Comment:
See `pyrosetta.rosetta.core.conformation.Residue.apply_transform_Rx_plus_v`

```
Docstring:
apply_transform_Rx_plus_v(self: pyrosetta.rosetta.core.conformation.Residue, R: pyrosetta.rosetta.numeric.xyzMatrix_double_t, v: pyrosetta.rosetta.numeric.xyzVector_double_t) -> None

Applies a transform of the form Rx + v, where R is a rotation
 matrix, V is a vector, and x is the original position in xyz space

C++: core::conformation::Residue::apply_transform_Rx_plus_v(class numeric::xyzMatrix<double>, class numeric::xyzVector<double>) --> void
Type:      method
```

Since the same transform is being applied to all of the coordinates in the `Pose`, you could do the following:
```python
def xyzMatrixFromNumpy(np_arr):
    rot = pyrosetta.rosetta.numeric.xyzMatrix_double_t(0.)
    rot.xx = np_arr[0, 0]
    rot.xy = np_arr[0, 1]
    rot.xz = np_arr[0, 2]

    rot.yx = np_arr[1, 0]
    rot.yy = np_arr[1, 1]
    rot.yz = np_arr[1, 2]

    rot.zx = np_arr[2, 0]
    rot.zy = np_arr[2, 1]
    rot.zz = np_arr[2, 2]

rot = xyzMatrixFromNumpy(xform[:3, :3])
trans = pyrosetta.rosetta.numeric.xyzVector_double_t(*xform[:3, 3])

for rsd in self.residues:
    rsd.apply_transform_Rx_plus_v(rot, trans)
```

Similar functionality to the `xyzMatrixFromNumpy` function above is provided in the `pyrosetta.toolbox.numpy_utils` module, but it's a little sketchy and could stand some improvement.

Reply:
Was not aware of this method. Thanks for letting me know. ✔️ ✔️

Reply:
Resolved by 76d3eb6.